### PR TITLE
Remove cdi dependecy from virtwrap API schema

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
### What this PR does
Drops the CDI dependency in the virtwrap schema.

This is one step towards dropping the dependency of kubevirt API on the CDI API, but the dependency still exists.

Fixes https://issues.redhat.com/browse/CNV-32043

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 
- https://issues.redhat.com/browse/CNV-32043
- https://redhat-internal.slack.com/archives/C017V3R4M08/p1692078453973249

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/converter/converter.go [hunk 1]
```diff
@@ -233,7 +233,7 @@ func Convert_v1_Disk_To_api_Disk(c *ConverterContext, diskDevice *v1.Disk, disk
 		}
 		volumeStatus, ok := volumeStatusMap[diskDevice.Name]
 		if ok && volumeStatus.PersistentVolumeClaimInfo != nil {
-			disk.FilesystemOverhead = volumeStatus.PersistentVolumeClaimInfo.FilesystemOverhead
+			disk.FilesystemOverhead = (*string)(volumeStatus.PersistentVolumeClaimInfo.FilesystemOverhead)
```
Comment (xpivarc) [inline pkg/virt-launcher/virtwrap/converter/converter.go:236]: Why don't we change the API as well?

File: pkg/virt-launcher/virtwrap/manager_test.go [hunk 2]
```diff
@@ -2885,7 +2885,8 @@ var _ = Describe("Manager helper functions", func() {
 		})
 
 		DescribeTable("should return error when", func(createDisk func() api.Disk) {
-
+			_, ok := possibleGuestSize(createDisk())
```
Comment (xpivarc) [inline pkg/virt-launcher/virtwrap/manager_test.go:2886]: +1

File: pkg/virt-launcher/virtwrap/manager.go [hunk 3]
```diff
@@ -893,6 +893,11 @@ func possibleGuestSize(disk api.Disk) (int64, bool) {
 		log.DefaultLogger().Reason(err).Error("Failed to parse filesystem overhead as float")
 		return 0, false
 	}
+	if filesystemOverhead < 0 || filesystemOverhead > 1 {
+		log.DefaultLogger().Errorf("Invalid filesystem overhead %f (must be between 0 and 1)", filesystemOverhead)
+		return 0, false
+	}
+
```
Comment (xpivarc) [inline pkg/virt-launcher/virtwrap/manager.go:900]: Note: This validation was not used before. Why do we handle the validation so deep in the stack?

File: pkg/virt-launcher/virtwrap/BUILD.bazel [hunk 4]
```diff
@@ -113,7 +113,6 @@ go_test(
         "//vendor/k8s.io/apimachinery/pkg/apis/meta/v1:go_default_library",
         "//vendor/k8s.io/apimachinery/pkg/types:go_default_library",
         "//vendor/k8s.io/utils/pointer:go_default_library",
-        "//vendor/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1:go_default_library",
```
Comment (xpivarc) [inline pkg/virt-launcher/virtwrap/BUILD.bazel:116]: Please inline 2369dd07f70569cd8352f5bd1ef778c8732b076c to respective commits.

File: pkg/virt-launcher/virtwrap/converter/converter.go (hunk 1 above)
Comment (EduardGomezEscandell) [reply pkg/virt-launcher/virtwrap/converter/converter.go]: Done :+1: 

I also created a Percent type alias to keep the same type safety while dropping the dependency.

File: pkg/virt-launcher/virtwrap/manager.go (hunk 3 above)
Comment (EduardGomezEscandell) [reply pkg/virt-launcher/virtwrap/manager.go:900]: Right before this we parse it from a string to a float, and I figured that it's easier to validate it when it is a float.

This way we reject out-of-range values in the same place we reject non-numerical values.

I can move it elsewhere if that's preferable.

Comment (xpivarc) [reply pkg/virt-launcher/virtwrap/manager.go:900]: I don't see a reason for the validation to be here (generally). I appreciate if we have a look at it in the follow up

**Review by xpivarc — COMMENTED** [review]
/approve

**Pull request merged**